contrib/babelfishpg_tsql/src/catalog.c

```diff
@@ -3250,3 +3250,76 @@ alter_guest_schema_for_db (const char *dbname)
 	table_endscan(tblscan);
 	table_close(bbf_authid_user_ext_rel, RowExclusiveLock);
 }
+
+/*
+ * Update the owner of a database in the catalog.
+ */
+void
+update_db_owner(const char *new_owner_name, const char *db_name)
+{
+	volatile 		Relation sysdatabases_rel;
+	TupleDesc		sysdatabases_rel_descr;
+	ScanKeyData		key;
+	HeapTuple		tuple, db_found;
+	TableScanDesc	tblscan;
+		
+	Datum		values[SYSDATABASES_NUM_COLS];
+	bool		nulls[SYSDATABASES_NUM_COLS];
+	bool		replaces[SYSDATABASES_NUM_COLS];
+
+	/* Do not allow changes to system databases. */
+	/* Note: T-SQL allows changing ownership of msdb. */
+	if ( (strlen(db_name) == 6 && (strncmp(db_name, "master", 6) == 0)) ||
+		 (strlen(db_name) == 6 && (strncmp(db_name, "tempdb", 6) == 0))
+	    )
+	{
+		ereport(ERROR,
+				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+				 errmsg("Cannot change the owner of the master, model, tempdb or distribution database.")));
+	}
+
+	/* Find the database */
+	sysdatabases_rel = table_open(sysdatabases_oid, RowExclusiveLock);
+	sysdatabases_rel_descr = RelationGetDescr(sysdatabases_rel);	
+
+	ScanKeyInit(&key,
+				Anum_sysdatabases_name,
+				BTEqualStrategyNumber, F_TEXTEQ,
+				CStringGetTextDatum(db_name));
```

Comment:
Do we store db_name in lowercase? if so then make sure that we pass downcases db_name here

Reply:
yes, it is stored in lowercase. the db name is downcased in makeChangeDbOwnerStatement()